unuran_perf.py

```diff
@@ -0,0 +1,421 @@
+    def dpdf(self, x):
+        if x == 0:
+            return 0
+        x = np.asarray(x)
+        first = (2*self.m-1)*x**(2*self.m-2) * np.exp(-self.m * x**2)
+        second = -2 * self.m * x * self.pdf(x)
+        return first + second
+    def dlogpdf(self, x):
+        if x == 0:
+            return np.nan
+        x = np.asarray(x)
+        return (2*self.m-1)/x - 2*self.m*x
+    def cdf(self, x):
+        if x == 0:
+            return 0
+        x = np.asarray(x)
+        return stats.nakagami._cdf(x, self.m)
+    def __repr__(self):
+        return f'nakagami({self.m})'
+
+
+# Studentized Range Distribution
+class studentized_range310:
+    scipy_dist = stats.studentized_range(3.0, 10.0)
+    numpy_dist = None
+    def __init__(self):
+        self.mode = optimize.minimize_scalar(lambda x: -self.pdf(x)).x
+        self.area = 1
+        self.domain = (-np.inf, np.inf)
+    def pdf(self, x):
+        return stats.studentized_range._pdf(x, 3.0, 10.0)
+    def logpdf(self, x):
+        return stats.studentized_range._logpdf(x, 3.0, 10.0)
+    def cdf(self, x):
+        return stats.studentized_range._cdf(x, 3.0, 10.0)
+    def __repr__(self):
+        return 'studentized_range(3, 10)'
+
+
+# Ksone distribution
+class ksone:
+    scipy_dist = stats.ksone(1000)
+    numpy_dist = None
+    def __init__(self):
+        self.mode = optimize.minimize_scalar(lambda x: -self.pdf(x), method='bounded', bounds=(0, 1)).x
+        self.area = 1
+        self.domain = (0, 1)
+    def pdf(self, x):
+        return stats.ksone._pdf(x, 1000)
+    def logpdf(self, x):
+        return stats.ksone._logpdf(x, 1000)
+    def cdf(self, x):
+        return stats.ksone._cdf(x, 1000)
+    def __repr__(self):
+        return 'ksone(1000)'
+
+
+# KStwo distribution
+class kstwo:
+    scipy_dist = stats.kstwo(10)
+    numpy_dist = None
+    def __init__(self):
+        self.mode = optimize.minimize_scalar(lambda x: -self.pdf(x), method='bounded', bounds=self.scipy_dist.support()).x
+        self.area = 1
+        self.domain = self.scipy_dist.support()
+    def pdf(self, x):
+        return stats.kstwo._pdf(x, 10)
+    def logpdf(self, x):
+        return stats.kstwo._logpdf(x, 10)
+    def cdf(self, x):
+        return stats.kstwo._cdf(x, 10)
+    def __repr__(self):
+        return 'kstwo(10)'
+
+
+allcontdists = [
+    kstwo(),
+    ksone(),
+    studentized_range310(),
+    nakagami(),
+    invgamma(),
+    genexpon(),
+    gausshyper(),
+    gennorm3(),
+    gamma2(),
+    stdnorm(),
+    beta23()
+]
+
+
+methods = [
+    'TransformedDensityRejection',
+    'AutomaticRatioOfUniforms',
+    'AdaptiveRejectionSampling',
+    'NumericalInverseHermiteUNURAN',
+    'NumericalInverse',
+    'NaiveRatioOfUniforms[mode]',
+    'NumericalInversePolynomial',
+    'SimpleRatioOfUniforms[mode, area]',
+    'SimpleSetupRejection[mode, area]',
+    'PiecewiseConstantHatsTable[mode, area]'
+]
+
+
+def get_rng(methodname, dist):
+    # parse the method string
```

Comment:
This was just a toy parser that I wrote initially. Need to change this to something that allows us to change the default parameters of the methods more flexibly. Maybe, we can use `functools.partial`.